sisl/viz/plots/geometry.py

```diff
@@ -445,7 +465,77 @@ def xyz(coeffs):
         if unique:
             points = np.unique(points, axis=0)
 
-        return self.geometry.origo + np.array([xyz(coeffs) for coeffs in points])
+        return np.array([xyz(coeffs) for coeffs in points])
+
+    @classmethod
+    def _projected_1Dcoords(cls, geometry, xyz=None, axis="x"):
+        """
+        Moves the 3D positions of the atoms to a 2D supspace.
+
+        In this way, we can plot the structure from the "point of view" that we want.
+
+        Parameters
+        ------------
+        geometry: sisl.Geometry
+            the geometry for which you want the projected coords
+        xyz: array-like of shape (natoms, 3), optional
+            the 3D coordinates that we want to project.
+            otherwise they are taken from the geometry. 
+        axis: {0,1,2, "x", "y", "z", "a", "b", "c"} or array-like of shape 3, optional
+            the direction to be displayed along the X axis. 
+            If it's an int, it will interpreted as the index of the cell axis.
```

Comment:
In your `_sanitize_axis` routine you interpret (0, 1, 2) as Cartesian coordinates?
I was confused whether `cell` meant lattice or Cartesian cell?
Please clarify?

Reply:
This is an old docstring that should have been changed. Initially `0` meant the first lattice vector, but since I realised that it was strange that users needed to project along a lattice vector I decided to change it. Now, as you understood, `0` is the same as `"x"` as it's always more comfortable to write integers than to write strings. Any opinion on whether this is fine or not?

Reply:
My other implementation here `sisl/utils/misc.py:direction` could solve this kind of easily. I would suspect integer indices to respect lattice vectors. Typing an `x` vs. a `0` is not a problem I think

Reply:
Hmm ok, the problem is that it's not an `x`, it's an `"x"` :) And when you do it lots of times it becomes annoying to write the `"`.

> I would suspect integer indices to respect lattice vectors.

Yeah me too, that's why this was the initial idea. But for the representations I can't see many cases where you would want to project the lattice vectors into the plot axes to visualize real space data.

Reply:
But isn't this in the GUI drop-down? I think `"x"` is fine to force people to do. It makes the code readable. Integers are not really good either. Better to force good habits than remove 2 characters. :)

Perhaps now there isn't much reason, but perhaps later...

Reply:
I have multiple times found that what makes it easier isn't always what users expect, and once it is in, it is hard to change. Always think in terms of functionality and what one would expect (without reading the documentation). Then if there is an ambiguity it can be discussed.

Reply:
Yes in the graphical interfaces this is clearly not a problem. It is also not a problem in scripts. But it is a problem in notebooks or other interactive environments that are very dynamic. In those cases you are changing code really fast and the displayed axes is a very recurrent thing to change.

I don't know, maybe `[0,1,2,3,4,5]` could stand for `["a", "b", "c", "x", "y", "z"]`? This would not cause problems for people who haven't read the documentation (they wouldn't expect `3,4,5` to be accepted), but allow people who have read it to be faster.

Reply:
I would really advocate in not doing this! Once you have typed the quotation marks you are only going to change x->a or a->x. So not a problem. Using your suggested approached is even more error prone, who would know/guess this combination of numbers?  
Again, it should be intuitive above all, the feel will be much simpler.  
Another example would be pandas.DataFrame, many of the methods have so many possibilities that I have to look up the documentation even for simple things I thought I could do by memory. This is because it is not natural or intuitive. Is asking for an initial 2 characters really a problem?

I don't really get your point given the above; even if you would like to be very dynamic, you still have typed the quotation marks once, and subsequent changes are only 1 character anyway?

Reply:
Also remember that users are *always* less efficient and less knowledgeable than you, they are not familiar with every piece of the software. So it should be *easy* to use, not to be fast. I have made many mistakes in this also. ;)

Reply:
> I don't really get your point given the above; even if you would like to be very dynamic, you still have typed the quotation marks once, and subsequent changes are only 1 character anyway?

The most common changes I make are:
- Change from 2D to 3D.
- Plot more grids/geometries, then I have to write again the axes that I want.

Writing `axes=["x", "y", "z"]` is annoying compared to `axes=[0,1,2]` (I'm experiencing it myself when writing it here, for the xyz version you need to press/unpress shift 6 times)

> So it should be easy to use, not to be fast.

I understand this, but if the point of the software is to allow quick visualization, then I think "quick to use" is also a feature that one would expect.

I'm worried that people will just use `0,1,2` as shortcuts for `"x", "y", "z"` when they have an (almost) orthogonal cell, not because they want the projection of the axes, but because they want the cartesian representation and it's more convenient to write integers. This will then not behave in the same way when the cell is not orthogonal.

Reply:
I am still not convinced ;)  
For rapid changes, a temporary variable can make everything much easier:
```python
axes = 'xyz'
#axes = [0,1,2]
```
Allow `axes` argument to handle strings as a list, say: `axes='xyz'` this would be even more natural and just as short, agreed `['x', 'y', 'z']` is too much. The other is more natural!

Reply:
Hmm maybe the issues that I have could be solved by allowing multicharacter strings also? I mean `axes="xyz"` would be the same as `axes=["x", "y", "z"]`. I think my speed thrive would be satisfied with that :) hahahah

Reply:
> > So it should be easy to use, not to be fast.
> 
> I understand this, but if the point of the software is to allow quick visualization, then I think "quick to use" is also a feature that one would expect.

I really don't think users are in this category ;)
Only very high level users will be irritated about this.

> I'm worried that people will just use `0,1,2` as shortcuts for `"x", "y", "z"` when they have an (almost) orthogonal cell, not because they want the projection of the axes, but because they want the cartesian representation and it's more convenient to write integers. This will then not behave in the same way when the cell is not orthogonal.

That is fine, there has to be rules. And you and I both thought that the lattice vectors were natural for integer indices, and so a large fraction of users will probably also have that thought. Of course there will be users that think the other way around, but regardless you'll get some users who don't expect the outcome of the arguments.

Reply:
Does this solution make sense to you?

Reply:
Ok, I see that we commented the same at the same time, then great!

Reply:
yes, great. I think this is better. Also consistency in sisl and elsewhere is key, I use integers as lattice vector matches.